cmake/ada-flags.cmake

```diff
@@ -50,7 +50,6 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
-set(CMAKE_MACOSX_RPATH OFF)
```

Comment:
Why did we remove this? @lemire

Reply:
By default, under macOS, CMake sets the [MACOSX_RPATH](https://cmake.org/cmake/help/latest/prop_tgt/MACOSX_RPATH.html#prop_tgt:MACOSX_RPATH) property on all targets. If you do it manually, you should set CMAKE_MACOSX_RPATH to false. It appears that we do not set MACOSX_RPATH, so we either need to do so, or to leave CMAKE_MACOSX_RPATH to its default value (ON).

https://cmake.org/cmake/help/latest/variable/CMAKE_MACOSX_RPATH.html

Note that this is not normally an issue because we rarely use dynamic libraries under macOS.